lib/oli/rendering/group/html.ex

```diff
@@ -18,6 +18,16 @@ defmodule Oli.Rendering.Group.Html do
     ]
   end
 
+  def group(%Context{} = context, next, params) do
+
+    id = Map.get(params, "id", UUID.uuid4())
+
+    params = Map.put(params, "id", id)
+    |> Map.put("purpose", "none")
```

Comment:
We should handle the `purpose` as the id. I mean, we can already have it in the params and use it from there, right?

Reply:
Ah yes, that is a great catch.  If only the id is missing this will reset the purpose